packages/flutter/lib/src/rendering/viewport.dart

```diff
@@ -753,7 +817,14 @@ abstract class RenderViewportBase<ParentDataClass extends ContainerParentDataMix
   }
 
   @override
-  RevealedOffset getOffsetToReveal(RenderObject target, double alignment, { Rect? rect }) {
+  RevealedOffset getOffsetToReveal(
+    RenderObject target,
+    double alignment, {
+    Rect? rect,
+    Axis? axis,
+  }) {
+    // One dimensional viewport has only one axis, override if set.
+    axis = this.axis;
```

Comment:
It was an assertion earlier, but I changed it. Not sure which is better. I added docs saying in 1D it would be ignored since it is meant to be used in 2D.

Reply:
I will probably use a assert. It is hard to say whether this will be used in the future even for 1D case, better to make it stricter now.

Reply:
Ok! I will change these back to asserts. Good idea! That will make the use easier too in your comment above about the documentation.